### PR TITLE
Add person name to PersonIsBannedFromSite error (#3786)

### PR DESCRIPTION
**Pull request by Freakazoid182**
As requested in #3786 

I mentioned in the issue that there might be a concern around logging possible PII data in logs. I'm not sure if this a concern here, would be good to get confirmed.

### CODE REVIEW

File: crates/apub/src/activities/mod.rs [hunk 1]
```diff
@@ -97,7 +97,9 @@ pub(crate) async fn verify_person_in_community(
 ) -> Result<(), LemmyError> {
   let person = person_id.dereference(context).await?;
   if person.banned {
-    return Err(LemmyErrorType::PersonIsBannedFromSite)?;
+    return Err(LemmyErrorType::PersonIsBannedFromSite(
+      person.name.to_owned(),
```
Comment (Nutomic) [inline crates/apub/src/activities/mod.rs:101]: person.name is not unique, there can be multiple users with the same name on different instances. Better to use `person.actor_id`

**Comment by Nutomic** [comment]
Whats the concrete use case for this? Its not clear to me from the issue.

File: crates/apub/src/activities/mod.rs (hunk 1 above)
Comment (Freakazoid182) [reply crates/apub/src/activities/mod.rs:101]: I was considering this might be better. Good catch

**Comment by Freakazoid182** [comment]
> Whats the concrete use case for this? Its not clear to me from the issue.

@urda, I'm sure you have a good use case in mind? :)

**Comment by urda** [comment]
> > Whats the concrete use case for this? Its not clear to me from the issue.
> 
> 
> 
> @urda, I'm sure you have a good use case in mind? :)

Sure.

@Nutomic @Freakazoid182 the current message just informs us a blocked user has attempted something. If we're taking the time to log that, it may be helpful to *know who* made the request. Adding this info makes the log message:

- worth storing
- worth evaluating

It can also be helpful to spot any issues or unforeseen situations, since you'll *know* what user triggered it.

File: crates/apub/src/objects/private_message.rs [hunk 2]
```diff
@@ -107,7 +107,9 @@ impl Object for ApubPrivateMessage {
     check_apub_id_valid_with_strictness(note.id.inner(), false, context).await?;
     let person = note.attributed_to.dereference(context).await?;
     if person.banned {
-      return Err(LemmyErrorType::PersonIsBannedFromSite)?;
+      return Err(LemmyErrorType::PersonIsBannedFromSite(
+        person.id.to_string(),
```
Comment (Nutomic) [inline crates/apub/src/objects/private_message.rs:111]: This is a numeric id, not really helpful. Use `person.actor_id` which is like `https://lemmy.ml/u/example` which is easy to understand.

**Review by Nutomic — APPROVED** [review]

**Pull request merged**